### PR TITLE
[GEOS-9214] GeoJSON complex encoder encodes separate FeaturePropertyType as a single array

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/GeoJsonOutputFormatWfsTest.java [hunk 1]
```diff
@@ -91,6 +91,7 @@ public void testGetGeoJsonResponseWfs20() throws Exception {
                 getAsJSON(
                         "wfs?request=GetFeature&version=2.0"
                                 + "&typenames=st_gml32:Station_gml32&outputFormat=application/json");
+        print(response);
```
Comment (nmco) [inline src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/GeoJsonOutputFormatWfsTest.java]: Leftover?

File: src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/GeoJsonOutputFormatWfsTest.java [hunk 2]
```diff
@@ -180,7 +189,6 @@ public void testOneDimensionalEncoding() throws Exception {
     public void testNestedFeatureEncoding() throws Exception {
         String path = "wfs?request=GetFeature&typename=gsml:Borehole&outputFormat=json";
         JSON json = getAsJSON(path);
-        print(json);
```
Comment (nmco) [inline src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/GeoJsonOutputFormatWfsTest.java]: Thx.

**Comment by nmco** [comment]
Looks good to me, thx @aaime. Will this be back-ported to 2.15.x?

**Comment by aaime** [comment]
Checked, to backport this the following also need to be backported:
* [GEOS-9179](https://github.com/geoserver/geoserver/commit/7a4eca58535f2a17865ecef02105d26d0c3f1a81#diff-4e0792e51b036935d54e2fb651e0ce7e)
* [GEOS-9180](https://github.com/geoserver/geoserver/commit/8e5446cd76aeac8bf50695cbb3a301f7b6535d6d#diff-4e0792e51b036935d54e2fb651e0ce7e)
* [GEOS-9181](https://github.com/geoserver/geoserver/commit/95c7e892c08ee3f53010713038f50e6aa9209d26#diff-4e0792e51b036935d54e2fb651e0ce7e)
* [GEOS-9204](https://github.com/geoserver/geoserver/commit/516f4e5367a7506b7e4a9d4c968fea5a695df33b#diff-4e0792e51b036935d54e2fb651e0ce7e)
* [GEOS-9205](https://github.com/geoserver/geoserver/commit/5ca877036920c268f9f96417dfb53953797a3fe8#diff-4e0792e51b036935d54e2fb651e0ce7e)

**Pull request merged**